components/clp-package-utils/clp_package_utils/scripts/start_clp.py

```diff
@@ -1136,195 +31,38 @@ def main(argv):
-
-        # Exit early if no components to start
-        if len(components_to_start) == 0:
-            logger.error(f"{target} not available with current configuration")
-            return -1
-
-        # Validate and load necessary credentials
-        if target in (
-            ALL_TARGET_NAME,
-            CONTROLLER_TARGET_NAME,
-            DB_COMPONENT_NAME,
-            GARBAGE_COLLECTOR_COMPONENT_NAME,
-            COMPRESSION_SCHEDULER_COMPONENT_NAME,
-            QUERY_SCHEDULER_COMPONENT_NAME,
-            WEBUI_COMPONENT_NAME,
-        ):
-            validate_and_load_db_credentials_file(clp_config, clp_home, True)
-        if target in (
-            ALL_TARGET_NAME,
-            CONTROLLER_TARGET_NAME,
-            QUEUE_COMPONENT_NAME,
-            COMPRESSION_SCHEDULER_COMPONENT_NAME,
-            QUERY_SCHEDULER_COMPONENT_NAME,
-            COMPRESSION_WORKER_COMPONENT_NAME,
-            QUERY_WORKER_COMPONENT_NAME,
-        ):
-            validate_and_load_queue_credentials_file(clp_config, clp_home, True)
-        if target in (
-            ALL_TARGET_NAME,
-            CONTROLLER_TARGET_NAME,
-            REDIS_COMPONENT_NAME,
-            COMPRESSION_SCHEDULER_COMPONENT_NAME,
-            QUERY_SCHEDULER_COMPONENT_NAME,
-            COMPRESSION_WORKER_COMPONENT_NAME,
-            QUERY_WORKER_COMPONENT_NAME,
-        ):
-            validate_and_load_redis_credentials_file(clp_config, clp_home, True)
-        if target in (
-            ALL_TARGET_NAME,
-            COMPRESSION_WORKER_COMPONENT_NAME,
-        ):
-            validate_logs_input_config(clp_config)
-        if target in (
-            ALL_TARGET_NAME,
-            COMPRESSION_WORKER_COMPONENT_NAME,
-            QUERY_WORKER_COMPONENT_NAME,
-            GARBAGE_COLLECTOR_COMPONENT_NAME,
-        ):
-            validate_output_storage_config(clp_config)
-        if target in (
-            ALL_TARGET_NAME,
-            CONTROLLER_TARGET_NAME,
-            GARBAGE_COLLECTOR_COMPONENT_NAME,
-        ):
-            validate_retention_config(clp_config)
+        validate_and_load_db_credentials_file(clp_config, clp_home, True)
+        validate_and_load_queue_credentials_file(clp_config, clp_home, True)
+        validate_and_load_redis_credentials_file(clp_config, clp_home, True)
+        validate_logs_input_config(clp_config)
+        validate_output_storage_config(clp_config)
+        validate_retention_config(clp_config)
 
         clp_config.validate_data_dir()
         clp_config.validate_logs_dir()
         clp_config.validate_aws_config_dir()
+
+        # Create necessary directories
+        clp_config.data_directory.mkdir(parents=True, exist_ok=True)
+        clp_config.logs_directory.mkdir(parents=True, exist_ok=True)
+        clp_config.archive_output.get_directory().mkdir(parents=True, exist_ok=True)
+        clp_config.stream_output.get_directory().mkdir(parents=True, exist_ok=True)
```

Comment:
The log of this try-except block doesn't make sense if one of these calls fails (fail to load doesn't imply anything about creating directories). Can we move the calls into their own try-except with a more appropriate log?